tests/certification/pubsub/azure/servicebus/servicebus_test.go

```diff
@@ -0,0 +1,346 @@
+		return func(ctx flow.Context) error {
+			// prepare the messages
+			messages := make([]string, numMessages)
+			for i := range messages {
+				messages[i] = fmt.Sprintf("partitionKey: %s, message for topic: %s, index: %03d, uniqueId: %s", metadata[messageKey], topicName, i, uuid.New().String())
+			}
+
+			// add the messages as expectations to the watchers
+			for _, messageWatcher := range messageWatchers {
+				messageWatcher.ExpectStrings(messages...)
+			}
+
+			// get the sidecar (dapr) client
+			client := sidecar.GetClient(ctx, sidecarName)
+
+			// publish messages
+			ctx.Logf("Publishing messages. sidecarName: %s, topicName: %s", sidecarName, topicName)
+
+			var publishOptions dapr.PublishEventOption
+
+			if metadata != nil {
+				publishOptions = dapr.PublishEventWithMetadata(metadata)
+			}
+
+			for _, message := range messages {
+				ctx.Logf("Publishing: %q", message)
+				var err error
+
+				if publishOptions != nil {
+					err = client.PublishEvent(ctx, pubsubName, topicName, message, publishOptions)
+				} else {
+					err = client.PublishEvent(ctx, pubsubName, topicName, message)
+				}
+				require.NoError(ctx, err, "error publishing message")
+			}
+			// Wait for double the TTL after sending the last message.
+			time.Sleep(time.Second * 20)
+			return nil
+		}
+	}
+
+	testDefault := func(ctx flow.Context) error {
+		ports, _ := dapr_testing.GetFreePorts(6)
+		grpcPortS1 := ports[0]
+		httpPortS1 := ports[1]
+		profilePortS1 := ports[2]
+		sidecar := sidecar.New(sidecarName5, embedded.WithComponentsPath("./components/consumer5"),
+			embedded.WithoutApp(),
+			embedded.WithDaprGRPCPort(grpcPortS1),
+			embedded.WithDaprHTTPPort(httpPortS1),
+			embedded.WithProfilePort(profilePortS1),
+			runtime.WithPubSubs(
+				pubsub_loader.New("azure.servicebus", func() pubsub.PubSub {
+					return pubsub_servicebus.NewAzureServiceBus(log)
+				}),
+			),
+			runtime.WithSecretStores(secretstores_loader.New("local.env", func() secretstores.SecretStore {
+				return secretstore_env.NewEnvSecretStore(log)
+			})))
+		sidecar.Start(ctx)
+		return nil
+	}
+
+	assertMessages := func(timeout time.Duration, messageWatchers ...*watcher.Watcher) flow.Runnable {
+		return func(ctx flow.Context) error {
+			// assert for messages
+			for _, m := range messageWatchers {
+				m.Assert(ctx, 25*timeout)
+			}
+
+			return nil
+		}
+	}
+
+	flow.New(t, "servicebus certification").
+
+		// Test : single publisher, multiple subscriber with their own consumerID
+		// Run subscriberApplication app1
+		Step(app.Run(appID1, fmt.Sprintf(":%d", appPort),
+			subscriberApplication(appID1, topicActiveName, consumerGroup1))).
+
+		// Run the Dapr sidecar with the eventhubs component 1, with permission at namespace level
+		Step(sidecar.Run(sidecarName1,
+			embedded.WithComponentsPath("./components/consumer1"),
```

Comment:
Please rename your component folders to give them more meaningful names. It should be more obvious which folder contains what.